.github/workflows/ci.yml

```
@@ -13,25 +13,49 @@ jobs:
         name: [ "Linux x64 (Ubuntu 20.04) - GCC 10" ]
         os: [ ubuntu-20.04 ]
         sanitize: [ false ]
+        prefix: [ "/usr/local" ]
+        shell: [ bash ]
+        sudo: [ sudo ]
         build:
-          - { cc: gcc, cxx: g++, linker: ld, shell: bash }
+          - { cc: gcc, cxx: g++, linker: ld }
         include:
           - name: "Linux x64 (Ubuntu 20.04) - Clang 10 with ASan and UBSan"
             os: ubuntu-20.04
             sanitize: true
-            build: { cc: clang-10, cxx: clang++-10, linker: ld.lld-10, shell: bash }
+            prefix: "/usr/local"
+            shell: bash
+            sudo: sudo
+            build: { cc: clang-10, cxx: clang++-10, linker: ld.lld-10 }
+
           - name: "macOS (10.15) - Xcode 12.3"
             os: macos-10.15
             sanitize: false
-            build: { cc: clang, cxx: clang++, linker: ld.lld, shell: bash }
+            prefix: "/usr/local"
+            shell: bash
+            sudo: sudo
+            build: { cc: clang, cxx: clang++, linker: ld.lld }
+
           - name: "macOS (10.15) - Xcode 12.3 - M1"
             os: macos-m1
             sanitize: false
-            build: { cc: clang, cxx: clang++, linker: ld.lld, shell: "/usr/bin/arch -arch arm64e /bin/bash -l {0}" }
+            prefix: "/Users/Administrator/vips"
+            shell: "/usr/bin/arch -arch arm64e /bin/bash -l {0}"
+            sudo: ""
+            build: { cc: clang, cxx: clang++, linker: ld.lld }
 
     defaults:
       run:
-        shell: ${{ matrix.build.shell }} 
+        shell: ${{ matrix.shell }}
+
+    env:
+      CC: ${{ matrix.build.cc }}
+      CXX: ${{ matrix.build.cxx }}
+      LD: ${{ matrix.build.linker }}
+      CPPFLAGS: -Wall
+      PREFIX: ${{ matrix.prefix }}
+      SUDO: ${{ matrix.sudo }}
+      PKG_CONFIG_PATH: "$PREFIX/lib/pkgconfig"
+      DYLD_LIBRARY_PATH: "$PREFIX/lib"
```

I just noticed that shell variables are not working here, see for example this CI log:
https://github.com/kleisauke/libvips/runs/3351200368#step:17:19

Though, this seems to work properly:
```suggestion
      PKG_CONFIG_PATH: "${{ matrix.prefix }}/lib/pkgconfig"
      DYLD_LIBRARY_PATH: "${{ matrix.prefix }}/lib"
```
(see https://github.com/kleisauke/libvips/runs/3351365557#step:17:19)